blueocean-events/src/main/java/io/jenkins/blueocean/events/PipelineEventListener.java

```diff
@@ -160,6 +168,22 @@ private Message newMessage(PipelineEventChannel.Event event, FlowNode flowNode,
                 message.set(PipelineEventChannel.EventProps.pipeline_step_name, stepNode.getDescriptor().getFunctionName());
             }
 
+            if (flowNode instanceof StepAtomNode) {
+                boolean pausedForInputStep = PipelineNodeUtil
+                    .isPausedForInputStep((StepAtomNode) flowNode, this.run.getAction(InputAction.class));
+                if (pausedForInputStep) {
+                    // Fire job event to tell we are paused
+                    // We will publish on the job channel
+                    try {
+                        PubsubBus.getBus().publish(new RunMessage(this.run)
+                            .setEventName(Events.JobChannel.job_run_paused)
+                        );
+                    } catch (MessageException e) {
+                        LOGGER.log(Level.WARNING, "Error publishing Run pause event.", e);
+                    }
+                }
+                message.set(PipelineEventChannel.EventProps.pipeline_is_paused, String.valueOf(pausedForInputStep));
```

Comment:
Agree with @tfennelly, we should transition to a single property with different values for the current run state, though I agree with @scherler `pipeline_step_is_paused` is more accurate, since only one branch could be waiting on input while others are executing.